src/dos/drive_overlay.cpp

```diff
@@ -540,10 +558,20 @@ std::unique_ptr<DOS_File> Overlay_Drive::FileCreate(const char* name,
 		return nullptr;
 	}
 
-	auto file = std::make_unique<OverlayFile>(
-	        name, path, file_handle, overlaydir, IsReadOnly());
+	DosDateTime dos_time  = {};
+	dos_time.date         = DOS_GetBiosDatePacked();
+	dos_time.time         = DOS_GetBiosTimePacked();
```

Comment:
Tip: You can use designated initialisers now that we got rid of old GCC versions:

```cpp
DosDateTime dos_time = {
    .date = DOS_GetBiosDatePacked(),
    .time = DOS_GetBiosTimePacked()
};
```

Also see 1b21f2f823f253e5d

Reply:
That's nifty.  I used that syntax in one other place but I didn't know you could put function calls in there.  It compiled though.

Reply:
> That's nifty. I used that syntax in one other place but I didn't know you could put function calls in there. It compiled though.

Sure you can. It's just syntactic sugar for what you've done previously, plus the really good thing about it is that it raises a warning if any of the struct members are left uninitialised.

This is super handy when adding new struct members down the road. With the old-school way, you'd get no warning about the newly added member not being initialised.

Reply:
Yeah, I like it.  I pushed it and it passed CI so I'm happy.  I think you could potentially run into problems if the order the functions get called in matters (in this case it doesn't).  Ex: If the order of the struct members differs from the order of your initialization list, I believe it will use the former.  Edge case though.  Most times we'll be better off using this.

Reply:
>  I think you could potentially run into problems if the order the functions get called in matters (in this case it doesn't).

Sure, if the function you're calling inline are not pure functions, anything is possible. That's a bit of broader problem though with non-pure functions.